930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/aws/AwsS3ArtifactDownloadHandler.java

```
@@ -119,6 +121,16 @@ private AwsConnectionDetails mapAwsConnectionDetails(
       awsAccessKey = credentials.getAccessKeyId();
       awsSecretKey = credentials.getSecretKey();
       awsToken = credentials.getToken();
+    } else if (awsConfigDecrypted.isUseIRSA()) {
+      AWSCredentialsProvider aWSCredentialsProvider =
+          awsHelperService.getCredentialsForIRSAonDelegate(awsConfigDecrypted);
+      awsAccessKey = aWSCredentialsProvider.getCredentials().getAWSAccessKeyId();
+      awsSecretKey = aWSCredentialsProvider.getCredentials().getAWSSecretKey();
+
+      if (aWSCredentialsProvider instanceof STSAssumeRoleSessionCredentialsProvider) {
```

what if it is not instance of STSAssumeRole? We should throw an exception in that case

---

also @namantalaycha10 please add a UT for the same

---

STSAssumeRole case is for AssumeCrossAccountRole, otherwise it will use access key and secret key from WebIdentityTokenCredentialsProvider